### DISCUSSION TITLE
ROS2 Bag Not Ingesting

### DISCUSSION THREAD

**Discussion by karlinsb** [foxglove/community#79] [Q&A]
We've been very held back by ROS2's bag playback functionality, the ability to pause and scrub in foxglove is a blessing. But some of our bag files are failing to import. We generate two types of bag files, one with all of the data from the vehicle, and another with the perception data stripped out for storage reasons. I can view the data from the stripped down files fine, but the full data bags with lidar and radar data are failing. I'm getting the 'Error initializing player" window, but no text in the box. I understand the current upstream limitations of ROS2 bags, but I'm only trying to view standard message types. Is there somewhere I can turn on more verbose error reporting to maybe get a better feel of what's freaking out Studio?
![Screenshot from 2021-10-19 16-03-15](https://user-images.githubusercontent.com/91696559/137982239-c5a30408-2604-427f-aaa6-26949ea3494b.png)

**Comment by amacneil** [comment]
Oh, I think this is due to not currently supporting ROS 2 bag files over 4gb. Is [this](https://github.com/foxglove/studio/issues/1957) your issue? (cc @jhurliman)

Unfortunately this is a limitation of loading sqlite in the browser. We definitely need a more helpful error message. We have some ideas for how to support >4gb ROS 2 bags in the desktop version, but our long term solution is to propose a [better file format for storing bag files in the next release of ROS 2](https://discourse.ros.org/t/evaluation-of-robotics-data-recording-file-formats/22588).

In the short term, would it be possible to split your files into <4gb chunks?

**Reply by karlinsb** [reply]
Ah, that is what happened. I found an older test where we rebagged around an incident of interest and found got this new exciting error message (below). pretty much all of our files are >4GB, but we were only really planning on using the desktop version anyway, so hopefully the hacky fix can be implemented easily.

> TypeError: Cannot convert null to a BigInt
    at BigInt (<anonymous>)
    at r.timeRange (file:///opt/Foxglove%20Studio/resources/app.asar/renderer/7544.js:2:6041)
    at s.timeRange (file:///opt/Foxglove%20Studio/resources/app.asar/renderer/7544.js:6:6429)
    at R.initialize (file:///opt/Foxglove%20Studio/resources/app.asar/renderer/8840.js:5:2741)
    at async w.initialize (file:///opt/Foxglove%20Studio/resources/app.asar/renderer/8517.js:1:1084)
